internal/resolver/dns/dns_resolver.go

```diff
@@ -47,8 +47,9 @@ var EnableSRVLookups = false
 
 var logger = grpclog.Component("dns")
 
-// A global to stub out in tests.
+// Globals to stub out in tests.
 var newTimer = time.NewTimer
+var newTimerDNSResRate = time.NewTimer
```

Comment:
Done.

Reply:
Personally I don't understand the reason to use a var block, except for a very long set of related global vars.  I find it doesn't improve readability, and in functions it adds 2 extra lines which is really something to be avoided IMO.

I'd say this is personal choice and should be recommended during reviews.

Reply:
On this particular line, it's really unfortunate to me that we can't use just _one_ NewTimer hooking global.  How much work would it take to do that?

Reply:
We have two global var blocks in this file, and outside of that we have multiple global vars defined in a non block. And in between this, there is an `init()` function and a `const` block. Sure, it is a personal choice. But I just feel vars are sprinkled all over the place while they can be nicely grouped in a block. And I agree this PR does not touch the other blocks, so making this a block does not help overall.

Reply:
A lotttttt of work @dfawley in regards to using one rather than two, as they are logically separate ideals (exponential backoff and simply the dns min res rate).

Reply:
This `var` here is just a `func (d time.Duration) *time.Timer`. Only the value of `d` passed to it depends on whether it comes from a min dns res rate or from an exponential backoff. Why is it a lot of work?

Reply:
It's a lot of work because you have to figure out what branch of logic it will take. Every test mocks the DNS res rate to 0 to not wait on it. I don't want to trigger exponential backoff on error conditions so I have a time.Hour return so the timer will never fire and it'll never trigger exponential backoff making tests actually deterministic. Making this a single global would require a few hours of manually going into each test and figuring out what branch of logic it takes, then mocking EACH CALL to new timer accordingly. It's a lot of work unless I'm misunderstanding it.

Reply:
Aren't all tests testing either one or the other behavior?  I may not be fully appreciating the complexity here, but it doesn't _seem_ that bad in theory.  I won't push this since I'm not certain, but it's worth a quick check IMO.

Reply:
No. Look at my exponential backoff test. It returns errors (backs off), then is ok (minDNSResRate at that point). Rewriting every single test around this concept doesn't make sense to me. I think it would take a whole day at least.

Reply:
Maybe add a TODO and in some other fixit we might revisit and see if we can make that happen. Or in hacktoberfest, some kind soul might do that.

Reply:
Added a TODO.